dataland-frontend/tests/e2e/fixtures/eutaxonomy/financials/EuTaxonomyDataForFinancialsFixtures.ts

```diff
@@ -7,126 +7,113 @@ import {
   type InsuranceKpis,
   type InvestmentFirmKpis,
 } from "@clients/backend";
-import { generateDatapointOrNotReportedAtRandom } from "@e2e/fixtures/common/DataPointFixtures";
 import { generateEuTaxonomyWithBaseFields } from "@e2e/fixtures/eutaxonomy/EuTaxonomySharedValuesFixtures";
-import { type ReferencedDocuments } from "@e2e/fixtures/FixtureUtils";
-import { randomPercentageValue } from "@e2e/fixtures/common/NumberFixtures";
-import { assertDefined } from "@/utils/TypeScriptUtils";
+import { DEFAULT_PROBABILITY, Generator } from "@e2e/utils/FakeFixtureUtils";
+import { generatePercentageValue } from "@e2e/fixtures/common/NumberFixtures";
 
 /**
- * Generates random insurance company KPIs
- * @param referencedReports he reports that can be referenced as data sources
- * @returns random insurance company KPIs
+ * Generates a single eutaxonomy-financials fixture
+ * @param undefinedProbability the probability (as number between 0 and 1) for "undefined" values in nullable fields
+ * @returns a random eutaxonomy-financials fixture
  */
-export function generateInsuranceKpis(referencedReports: ReferencedDocuments): InsuranceKpis {
-  const taxonomyEligibleNonLifeInsuranceActivities = randomPercentageValue();
-  return {
-    taxonomyEligibleNonLifeInsuranceActivities: generateDatapointOrNotReportedAtRandom(
-      taxonomyEligibleNonLifeInsuranceActivities,
-      referencedReports,
-    ),
-  };
+export function generateEuTaxonomyDataForFinancials(
+  undefinedProbability = DEFAULT_PROBABILITY,
+): EuTaxonomyDataForFinancials {
+  const dataGenerator = new EuFinancialsGenerator(undefinedProbability);
+  return dataGenerator.generateEuTaxonomyDataForFinancialsWithTypes();
 }
 
-/**
- * Generates random credit institution KPIs
- * @param referencedReports he reports that can be referenced as data sources
- * @returns random credit institution KPIs
- */
-export function generateCreditInstitutionKpis(referencedReports: ReferencedDocuments): CreditInstitutionKpis {
-  let tradingPortfolioAndInterbankLoans = undefined;
-  let interbankLoans = undefined;
-  let tradingPortfolio = undefined;
+export class EuFinancialsGenerator extends Generator {
+  financialServicesTypes: EuTaxonomyDataForFinancialsFinancialServicesTypesEnum[];
 
-  const singleOrDualField = faker.datatype.boolean();
-  if (singleOrDualField) {
-    tradingPortfolioAndInterbankLoans = randomPercentageValue();
-  } else {
-    interbankLoans = randomPercentageValue();
-    tradingPortfolio = randomPercentageValue();
+  constructor(undefinedProbability = DEFAULT_PROBABILITY) {
```

Comment:
It seems that this constructor is not really needed.
- You can access the `undefinedProbability` with `this.undefinedProbability` even without it, because you are in an extension of the Generator, right? At least it seems to work for me.
- And the assignment of values to `this.financialServicesTypes` can simply happen in line 27 via:
`financialServicesTypes = faker.helpers.arrayElements(
      Object.values(EuTaxonomyDataForFinancialsFinancialServicesTypesEnum),
  );`
  It seems that you can even omit the type of that variable then.

Reply:
agreed this can be simplified. Removed the constructor